### ISSUE TITLE
Metadata attribute doesn't generate swagger data for properties

### ISSUE THREAD

**Issue by BenasJacikas** [nihaue/TRex#11]
Example:

```
public class DocumentsController : ApiController 
{
    [Route("api/document/{id}"
    public IHttpActionResult Put (string id, [FromBody] Document document) 
    {
        return Ok(document);
    }
}

public class Document 
{
    [JsonProperty("fileName")]
    [Metadata("File name")]
    public string FileName { get; set; } 
}
```

Code above doesn't generate `x-ms-summary` (or any other information if that matters) for `FileName`, Flow/Powerapps support this functionality.

**Comment by BenasJacikas** [comment]
Ok, pinpointed issue to 
```
        public void Apply(Schema schema, SchemaRegistry schemaRegistry, Type type)
        {
            if (schema == null || type == null)
            {
                return;
            }

            applySchemaLookupForDynamicModels(schema, schemaRegistry, type);

            if (schema.properties == null)
            {
                return;
            }

            foreach (var propertyName in schema.properties.Keys)
            {
                var schemaProperty = schema.properties[propertyName];
                var propertyInfo = type.GetRuntimeProperties().FirstOrDefault(p => p.Name == propertyName);

                applyPropertyMetadata(schemaProperty, propertyInfo);
                applyCallbackUrl(schemaProperty, propertyInfo);
            }
        }
```

`var propertyInfo = type.GetRuntimeProperties().FirstOrDefault(p => p.Name == propertyName);` uses swagger generated property name which isn't always in the same "casing" (sorry don't know the correct term for this), for example my code is in PascalCase and json is in camelCase.

**Comment by nihaue** [comment]
Confirmed. I'll get that fixed up 👍

**Comment by nihaue** [comment]
Fix has been implemented in [v2.0.4-alpha of the nuget package](https://www.nuget.org/packages/TRex/2.0.4-alpha).
